### PR TITLE
Move code related to tpm2_checkquote into its own function

### PR DESCRIPTION
**Pull request by stefanberger**
This PR makes the check_quote function a bit easier to read by moving tpm2_checkquote related code into is own function.

### CODE REVIEW

File: keylime/tpm/tpm_main.py [hunk 1]
```diff
@@ -986,15 +986,15 @@ def __check_quote_c(self, pubaik, nonce, quoteFile, sigFile, pcrFile, hash_alg):
...
-    def check_quote(self, agent_id, nonce, data, quote, aikTpmFromRegistrar, tpm_policy={}, ima_measurement_list=None, allowlist={}, hash_alg=None, ima_keyring=None, mb_measurement_list=None, mb_refstate=None):
-        if hash_alg is None:
-            hash_alg = self.defaults['hash']
-
-        quoteFile = None
-        aikFile = None
-        sigFile = None
-        pcrFile = None
-
+    def do_tpm2_checkquote(self, aikTpmFromRegistrar, quote, nonce, hash_alg):
```
Comment (mpeters) [inline keylime/tpm/tpm_main.py:989]: Should this be named `_do_tpm2_checkquote` since it's not intended to be a public interface, right?

**Review by mpeters — CHANGES_REQUESTED** [review]

File: keylime/tpm/tpm_main.py (hunk 1 above)
Comment (stefanberger) [reply keylime/tpm/tpm_main.py]: I renamed it to `_tpm2_checkquote` now.

**Comment by maugustosilva** [comment]
LGTM, I just tested on my end-to-end CI and it passed

**Review by mpeters — APPROVED** [review]
lgtm

**Pull request merged**